cql/ManageRareAbnormality.cql

```diff
@@ -1089,32 +1126,42 @@ define RecommendationForHistologyResults:
     // I.1 Management of Histologic HSIL, not Further Specified or Qualified
     when (
       AgeInYears() >= 25 and AgeInYears() <= 50 and
+      AnyHistologyInterpretedAsUnspecifiedHsil and
       not RecentlyRespondedYesToFuturePregnancyConcernsQuestion and
-      MostRecentBiopsyReportWasWithinPastYear and
-      HistologyInterpretedAsUnspecifiedHsil
+      (
+        Collate.DateOfLastCervicalPrecancerTreatment is null or 
+        not (Collate.DateOfLastCervicalPrecancerTreatment occurs after Collate.MostRecentHsilBiopsy.date) or
+        Collate.DateOfLastCervicalPrecancerTreatment 7 years or more before Collate.MostRecentHsilBiopsy.date
+      ) and
+      not SecondMostRecentNegativeHsilSurveillanceResultWithin18Months
```

Comment:
Not sure - will have to check with SS/GM.

Reply:
I think I may have been looking at the wrong section of the L2. Looking now at section I.1 - recommendation for 25-50 y.o., it appears that the following clause was added:

> AND COUNT of ‘negative HSIL surveillance’ <= 18 months AFTER the "Histologic HSIL, unspecified" result < 2 

I believe the goal of this is so that patients who have already had 2 negative HSIL surveillance tests don't continue to get this recommendation, since they will get a recommendation from section I.3, which deals with patients who have chosen to undergo observation. I think we should add this back to the logic.